### ISSUE TITLE
[FEATURE REQUEST] Possibility to reset currentMode

### ISSUE THREAD

**Issue by HardcoreWR** [Woozinator/node-red-contrib-time-based-dimmer#10]
Hello,

My lights are turned on/off completely automatically based on how bright it is in the room.
Also dimlevel is currently completely automated.
However, it might be useful to override the automated dimlevel.
I have functionality already that if I make any changes by hand, the automation for the lights will be disabled until I tell it to start the automation.

It would be nice to send a msg.reset to the one-button-dimmer after I start the automation again so the next startCommand would trigger 'inc'.

If you need more information, please let me know

**Comment by woozar** [comment]
That is a nice feature request. I think it would even be better to allow to send "next":"inc" or "next": "dec" or something.
There is only one question: what hardware do you use to measure current brightness? 😂

**Comment by HardcoreWR** [comment]
Yes, sending a specific mode would be nice as well, gives you extra flexibility for other purposes.

I'm using Home Assistant for controlling my smart devices (NodeRED only for the automation side of it).
There's a nice project called ESPHome (https://esphome.io/), and that allow you to very easily attach al sorts of sensors and have it report to Home Assistant using an ESP32 or ESP8266.

I'm using an ESP32 with a TSL2561 ambient light sensor (https://esphome.io/components/sensor/tsl2561.html)
This reports the ambient light every 3s which is very nice, if I close the curtains my lights gradually turn on :) (and the other way around)

**Comment by woozar** [comment]
I implemented it. Just send 
{
  "next": "inc"
}
to the node, to reset the dimmer. If you do this while keeping the button pressed it will take effect immediately. 
Possible values: "inc" or "dec".

**Comment by HardcoreWR** [comment]
Hello,

I just tried this method but I don't get this working.
I'm using a function node and an inject node to verify but in both situation it's not changing the mode.

This is the debug message from my inject node:
`{"_msgid":"b13c2a44.61e818","topic":"","payload":{"next":"inc"}}`

Using Node-RED 1.0.4 and node-red-contrib-time-based-dimmer v0.8.1.
> 
> 
> I implemented it. Just send
> {
> "next": "inc"
> }
> to the node, to reset the dimmer. If you do this while keeping the button pressed it will take effect immediately.
> Possible values: "inc" or "dec".

**Comment by woozar** [comment]
If you send that packet and take a look at the context of the node, the "mode" should switch to "dec" so the next start command will invert the mode and hence start increasing the value.

**Comment by HardcoreWR** [comment]
That's what I was expecting, but it doesn't seem to change here.
Does the packet looks good to you?

**Comment by woozar** [comment]
Yes. I will take a look, later.

**Comment by HardcoreWR** [comment]
I kept on looking at the source for one-button-dimmer.ts and couldn't find anything wrong.
I then decided to take a look what's in my `node-red/node_modules/node-red-contrib-time-based-dimmer` directory and I noticed the whole part, where you check if `msg.payload` is an object, is missing.

I have never built any packages (or modules) to use with NodeRED so I don't know how that works (yet, I want to look at it later)
Maybe this pushes you into the right direction.
I'm using v0.8.1 of this module.
